lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -88,24 +88,23 @@ const rule = (primary, secondaryOptions, context) => {
 
 				const indexDuplicate = decls.findIndex((d) => d.prop.toLowerCase() === lowerProp);
 
-				if (indexDuplicate !== -1) {
-					const duplicateDecl = decls[indexDuplicate];
-					const duplicateValue = duplicateDecl ? duplicateDecl.value : '';
-					const duplicateImportant = duplicateDecl ? duplicateDecl.important : false;
-
-					if (ignoreDiffValues || ignorePrefixlessSameValues) {
-						// fails if duplicates are not consecutive
-						if (indexDuplicate !== decls.length - 1) {
-							if (context.fix) {
-								if (!important && duplicateImportant) {
-									decl.remove();
-								} else {
-									removePreviousDuplicate(decls, lowerProp);
-								}
+				if (indexDuplicate === -1) {
+					decls.push(decl);
+				}
 
-								return;
-							}
+				const duplicateDecl = decls[indexDuplicate];
+
+				if (!duplicateDecl) {
+					return;
+				}
+
+				const duplicateValue = duplicateDecl.value || '';
+				const duplicateImportant = duplicateDecl.important || false;
 
+				if (ignoreDiffValues || ignorePrefixlessSameValues) {
+					// fails if duplicates are not consecutive
+					if (indexDuplicate !== decls.length - 1) {
```

Comment:
suggestion

put `indexDuplicate !== decls.length - 1` in a const foo (dunno about the name but you ought to)
and use `!foo` at l178